webrender/src/render_task.rs

```diff
@@ -536,62 +554,55 @@ impl RenderTask {
     //           +---- This is stored as the input task to the primitive shader.
     //
     pub fn new_blur(
-        blur_std_deviation: f32,
-        src_task_id: RenderTaskId,
+        radius: f32,
+        mut src_task_id: RenderTaskId,
         render_tasks: &mut RenderTaskTree,
         target_kind: RenderTargetKind,
         clear_mode: ClearMode,
     ) -> Self {
-        // Adjust large std deviation value.
-        let mut adjusted_blur_std_deviation = blur_std_deviation;
-        let (blur_target_size, uv_rect_kind) = {
-            let src_task = &render_tasks[src_task_id];
-            (src_task.get_dynamic_size(), src_task.uv_rect_kind())
-        };
-        let mut adjusted_blur_target_size = blur_target_size;
-        let mut downscaling_src_task_id = src_task_id;
-        let mut scale_factor = 1.0;
-        while adjusted_blur_std_deviation > MAX_BLUR_STD_DEVIATION {
-            if adjusted_blur_target_size.width < MIN_DOWNSCALING_RT_SIZE ||
-               adjusted_blur_target_size.height < MIN_DOWNSCALING_RT_SIZE {
-                break;
-            }
-            adjusted_blur_std_deviation *= 0.5;
-            scale_factor *= 2.0;
-            adjusted_blur_target_size = (blur_target_size.to_f32() / scale_factor).to_i32();
-            let downscaling_task = RenderTask::new_scaling(
-                target_kind,
-                downscaling_src_task_id,
-                adjusted_blur_target_size,
-            );
-            downscaling_src_task_id = render_tasks.add(downscaling_task);
+        // Look up some info.
+        let full_size = render_tasks[src_task_id].get_dynamic_size();
+        let uv_rect_kind = render_tasks[src_task_id].uv_rect_kind();
+
+        // Determine the number of downsampling passes that we'll use.
+        let downsampling_passes = (30 as u32).saturating_sub((radius as u32).leading_zeros());
```

Comment:
number of downsample passes = 30 - (leading 0 bits of radius)?
where does 30 come from?